daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/parsers/SequenceParserBases.scala

```diff
@@ -194,6 +194,12 @@ abstract class SequenceParserBase(
                 pstate.mpstate.moveOverOneArrayIndexOnly()
               }
 
+              // If the emptyElementParsePolicy is set to treatAsAbsent, don't
+              // increment the occursIndex if the element is absent
+              if (resultOfTry != AbsentRep) {
```

Comment:
Does resultOfTry only come back AbsentRep if treatAsAbsent is set?
I'm not sure that's the case. If so then ok.

I expected this to test resultOfTry as you have, but also to inspect the property (which has to be captured onto the runtime data structures) to see if it is treatAsAbsent.

Reply:
I was assuming that if treatAsAbsent was not set that it would be EmptyRep instead which I think is the case? I did try to see if I could check the emptyElementParsePolicy at that point but I wasn't able to navigate the object tree to find it.

Reply:
Ok. I suspect you are right, and if tests aren't breaking on account of this, then we are almost certainly ok.